server/protocols/jmap/src/main/java/org/apache/james/jmap/model/Keyword.java

```diff
@@ -0,0 +1,188 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.model;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.mail.Flags;
+
+import org.apache.james.mailbox.FlagsBuilder;
+import org.apache.commons.lang.StringUtils;
+
+import com.github.steveash.guavate.Guavate;
+import com.google.common.base.CharMatcher;
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Objects;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableBiMap;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+
+public class Keyword {
+    public final static boolean FLAG_VALUE = true;
+    private final static int FLAG_NAME_MIN_LENTH = 1;
+    protected final static int FLAG_NAME_MAX_LENTH = 255;
+    private static final CharMatcher FLAG_NAME_PATTERN = CharMatcher.JAVA_LETTER_OR_DIGIT
+            .or(CharMatcher.is('$'));
+
+    public final static Keyword DRAFT = new Keyword("$Draft", FLAG_VALUE);
+    public final static Keyword SEEN = new Keyword("$Seen", FLAG_VALUE);
+    public final static Keyword FLAGGED = new Keyword("$Flagged", FLAG_VALUE);
+    public final static Keyword ANSWERED = new Keyword("$Answered", FLAG_VALUE);
+    public final static Keyword DELETED = new Keyword("$Deleted", FLAG_VALUE);
+    public final static Keyword RECENT = new Keyword("$Recent", FLAG_VALUE);
+
+    public static final ImmutableList<Keyword> UNSUPPORTED_KEYWORDS = ImmutableList.of(Keyword.RECENT, Keyword.DELETED);
+
+    public static final ImmutableBiMap<Flags, Keyword> IMAP_SYSTEM_FLAGS = ImmutableBiMap.<Flags, Keyword>builder()
```

Comment:
The type should be exactly: ImmutableBiMap<Flags**.Flag**, Keyword>

Because as is, it makes the code harder... We don't map to Flags but to a system flag.